task/jefe/src/dump.rs

```diff
@@ -0,0 +1,168 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Dump support for Jefe
+
+use humpty::{DumpAgent, DumpArea};
+use ringbuf::*;
+use task_jefe_api::DumpAgentError;
+use userlib::*;
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum Trace {
+    None,
+    Initialized,
+    GetDumpArea(u8),
+    Base(u32),
+    GetDumpAreaFailed(humpty::DumpError<()>),
+    ClaimDumpAreaFailed(humpty::DumpError<()>),
+    Claiming,
+    Dumping(usize, Option<u32>),
+    DumpArea(Result<Option<DumpArea>, humpty::DumpError<()>>),
+    DumpRegion(abi::TaskDumpRegion),
+    DumpRegionsFailed(humpty::DumpError<()>),
+    DumpStart(u32),
+    DumpReading(u32, usize, bool),
+    DumpRead(usize),
+    DumpDone(Result<(), humpty::DumpError<()>>),
+}
+
+ringbuf!(Trace, 8, Trace::None);
+
+pub fn initialize_dump_areas() -> Option<u32> {
+    let areas = humpty::initialize_dump_areas(
+        &crate::generated::DUMP_AREAS,
+        Some(0x1000),
+    );
+
+    ringbuf_entry!(Trace::Initialized);
+
+    areas
+}
+
+pub fn get_dump_area(
```

Comment:
`get_dump_area` and `claim_dump_area` would be cleaner if they accepted `base: u32` instead of `Option<u32>`, moving the `NoDumpAreas` error back into `jefe`'s `ServerImpl`.

Reply:
Agreed; done